hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/HddsVolume.java

```diff
@@ -114,9 +117,8 @@ public void createWorkingDir(String workingDirName,
       MutableVolumeSet dbVolumeSet) throws IOException {
     super.createWorkingDir(workingDirName, dbVolumeSet);
 
-    if (SchemaV3.isFinalizedAndEnabled(getConf())) {
-      createDbStore(dbVolumeSet);
-    }
+    // Create DB store for a newly formatted volume
```

Comment:
Maybe `isFinalized()` check is still needed here? If not, a newly added Volume after code upgrade and before finalize schema v3  will create a db?

Reply:
This is the expected behavior.  This case is the same as a new Ozone cluster case,  DB will be created without the need of schema V3 finalization.

Reply:
But then we are using part of the un-finalized feature logic before finalizeUpgrade, conceptually, if it is OK for non-rolling upgrade feature, I'm fine.

Reply:
Think about the new cluster case,  we should support create DB without running upgrade action, right?  If we put the DB creation in upgrade action,  how can we support a new schema V3 cluster?

Reply:
This is a good case, but it does not leads to remove the `isFinalized` check here, because in the freshly deployed new cluster, all layout features will be initialized to a finalized state since there is nothing to be upgraded, check `DatanodeLayoutStorage.getDefaultLayoutVersion()`.